public/app/core/components/RolePicker/UserRolePicker.tsx

```diff
@@ -53,15 +54,14 @@ export const UserRolePicker = ({
   width,
   isLoading,
 }: Props) => {
-  const [{ loading, value: appliedRoles = roles || [] }, getUserRoles] = useAsyncFn(async () => {
+  const [userRolesState, getUserRoles] = useAsyncFn(async () => {
     try {
-      if (roles) {
+      if (isEqual(roles, userRolesState.value)) {
```

Comment:
It does not work because initially, roles are undefined. So it's always equal at first rendering. So this small fix solves the problem:
```suggestion
      if (roles && isEqual(roles, userRolesState.value)) {
```